src/platform/forms-system/src/js/review/submit-states/ValidationError.jsx

```diff
@@ -1,24 +1,142 @@
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
 import Back from './Back';
 import ProgressButton from '../../components/ProgressButton';
-import PropTypes from 'prop-types';
 import { Column, Row } from 'platform/forms/components/common/grid';
 import ErrorMessage from 'platform/forms/components/common/alerts/ErrorMessage';
 import PreSubmitSection from 'platform/forms/components/review/PreSubmitSection';
 
-export default function ValidationError(props) {
-  const { appType, buttonText, formConfig, onBack, onSubmit, testId } = props;
+import { focusAndScrollToReviewElement } from '../../utilities/ui';
+import { openReviewChapter, setEditMode } from '../../actions';
+
+function ValidationError(props) {
+  const {
+    appType,
+    buttonText,
+    formConfig,
+    onBack,
+    onSubmit,
+    testId,
+    form,
+  } = props;
+
+  const errorRef = useRef(null);
+  const errors = form.formErrors?.errors || [];
+  const errorsLen = errors.length;
+  const [hadErrors, setHadErrors] = useState(false);
+
+  // error links need evaluation & testing before production
+  const renderErrors =
+    (errorsLen > 0 && formConfig.showReviewErrors?.()) || false;
+
+  useEffect(
+    () => {
+      // Move focus to legend
+      if (renderErrors && !errorRef.current.classList.contains('has-focused')) {
+        // initially focus on alert legend immediately above error links
+        errorRef.current.focus();
+        errorRef.current.classList.add('has-focused');
+      }
+    },
+    [renderErrors, errorRef],
+  );
+
+  let alert;
+  if (renderErrors) {
+    if (!hadErrors && errorsLen > 0) {
```

Comment:
I recommend splitting this branch into its own module. This will
1. Keep this component easier to read (I had to check the PR out and look at this file in my editor so I can get code folding; there's a _lot_ going on here)
    ```jsx
    const alert = renderErrors ? <NewErrorMessage .../> : <ErrorMessage .../>;
    ```
2. Make it easier to test smaller parts at a time

Reply:
I'm not sure what you mean by "splitting this branch into its own module"? Do you mean copy the `renderErrors` block into a separate file? Or do you mean open a new PR with just this code?

Reply:
Done. I created a `ErrorLinks` component.